cmd/traffic/cmd/agent/fwdstate.go

```diff
@@ -160,56 +313,19 @@ func (fs *fwdState) HandlePort(ctx context.Context, cepts []*manager.InterceptIn
 
 	// Review waiting intercepts
 	reviews := make([]*manager.ReviewInterceptRequest, 0, len(waiting))
-	for _, ii := range waiting {
-		switch {
-		case activeIntercept == nil || ii.Spec.Wiretap:
-			// This intercept is ready to be active
-			container := ii.Spec.ContainerName
-			if container == "" {
-				container = fs.container
-			}
-			cs := fs.containerStates[container]
-			if cs == nil {
-				reviews = append(reviews, &manager.ReviewInterceptRequest{
-					Id:                ii.Id,
-					Disposition:       manager.InterceptDispositionType_AGENT_ERROR,
-					Message:           fmt.Sprintf("No match for container %q", container),
-					MechanismArgsDesc: generateMechanismDescription(ii.Spec),
-				})
-				continue
-			}
-			if !ii.Spec.Wiretap {
-				// We can only have one active intercept that isn't a wiretap
-				activeIntercept = ii
-			}
-			reviews = append(reviews, &manager.ReviewInterceptRequest{
-				Id:                ii.Id,
-				Disposition:       manager.InterceptDispositionType_ACTIVE,
-				PodIp:             fs.PodIP().String(),
-				FtpPort:           int32(fs.FtpPort()),
-				SftpPort:          int32(fs.SftpPort()),
-				MountPoint:        cs.MountPoint(),
-				Mounts:            cs.Mounts().ToRPC(),
-				MechanismArgsDesc: generateMechanismDescription(ii.Spec),
-				Environment:       cs.Env(),
-			})
-		default:
-			// We already have an intercept in play, so reject this one.
-			chosenID := activeIntercept.Id
-			dlog.Infof(ctx, "Setting intercept %q as AGENT_ERROR; as it conflicts with %q as the current chosen-to-be-ACTIVE intercept", ii.Id, chosenID)
-			var msg string
-			if activeIntercept.Disposition == manager.InterceptDispositionType_ACTIVE {
-				msg = fmt.Sprintf("Conflicts with the currently-served intercept %q", chosenID)
-			} else {
-				msg = fmt.Sprintf("Conflicts with the currently-waiting-to-be-served intercept %q", chosenID)
-			}
-			reviews = append(reviews, &manager.ReviewInterceptRequest{
-				Id:                ii.Id,
-				Disposition:       manager.InterceptDispositionType_AGENT_ERROR,
-				Message:           msg,
-				MechanismArgsDesc: generateMechanismDescription(ii.Spec),
-			})
+
+	// Collect all intercepts that could potentially become active
+	candidateIntercepts := slices.Clone(waiting)
+
+	for i, ii := range candidateIntercepts {
+		if ii.Spec.Wiretap {
+			reviews = append(reviews, fs.processWiretapIntercept(ii))
+			continue
 		}
+
+		// Check for conflicts and process regular intercept
+		review := fs.processRegularIntercept(ctx, ii, i, active, candidateIntercepts, &activeIntercept)
```

Comment:
Thinking more about this. The concept of an `activeIntercept` is actually invalid now, because there can be several non-conflicting active intercepts on the same port/proto combination. Shouldn't this be used only for the one and only "global intercept" if there is such an intercept?